addons/account_qr_code_emv/models/res_bank.py

```diff
@@ -59,13 +59,14 @@ def _get_qr_code_vals_list(self, qr_method, amount, currency, debtor_partner, fr
         comment = structured_communication or free_communication or ''
         comment = re.sub(r'/[^ A-Za-z0-9_@.\/#&+-]+/g', '', remove_accents(comment))
         additional_data_field = self._get_additional_data_field(comment) if self.include_reference else None
+        amount_data = [(54, amount)] if amount else ()
```

Comment:
The _serialize method will remove the data if the value is None or ''. Therefore you could just change the amount to None if 0